src/fmeval/transforms/common.py

```diff
@@ -191,3 +195,76 @@ def __call__(self, record: Dict[str, Any]) -> Dict[str, Any]:
         avg = np.mean([record[input_key] for input_key in self.input_keys])
         record[self.output_key] = avg
         return record
+
+
+class BertScoreMax(Transform):
+    """This Transform augments its input record with the maximum BERTScore metric computed over
+    possible targets separated by a target_output_delimiter.
+    """
+
+    def __init__(
+        self,
+        target_output_keys: List[str],
+        model_output_keys: List[str],
+        output_keys: List[str],
+        allow_duplicate_input_keys: bool,
+        bertscore_model: Union[BertscoreHelperModel, ActorHandle],
+        target_output_delimiter: Optional[str] = "<OR>",
+    ):
+        """BertScoreMax initializer.
+        :param target_output_keys: The keys corresponding to target outputs.
+        :param model_output_keys: The keys corresponding to model outputs.
+        :param output_keys: The output keys for this Transform, which correspond
+            to the BERT scores that get computed.
+        :param allow_duplicate_input_keys: See docstring for SummarizationAccuracyMetric.
+        :param bertscore_model: A BertscoreHelperModel instance or a Ray actor handle for a BertscoreHelperModel.
+        :param split_function: This is essentially a function that allows us to convert our exists target output into
+            a list of possible targets.
+        """
+        super().__init__(
+            target_output_keys,
+            model_output_keys,
+            output_keys,
+            allow_duplicate_input_keys,
+            bertscore_model,
+            target_output_delimiter,
```

Comment:
I originally had a `split_function` parameter which was a function that converted the target_output into a list of possible targets instead of the `target_output_delimiter` in order to keep the component more generic in its use case. However, when implementing, I ran into an issue with ray in one of the tests. Will continue debugging and working on making this component more generic
<img width="1512" alt="Screenshot 2024-07-25 at 3 25 18 PM" src="https://github.com/user-attachments/assets/8a08d650-44bc-4723-9d89-240c27c3d6a2">